### PR TITLE
Updating todos

### PR DESCRIPTION
**Pull request by eddynaka**
Refs #942

## Changes
- Updating TODO places where we said to use IsRemote
- Updating tests

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs [hunk 1]
```diff
@@ -59,9 +59,7 @@ public void TraceContextFormatCanParseExampleFromSpec()
             Assert.Equal(ActivityTraceId.CreateFromString(TraceId.AsSpan()), ctx.TraceId);
             Assert.Equal(ActivitySpanId.CreateFromString(SpanId.AsSpan()), ctx.SpanId);
 
-            // TODO: when ActivityContext supports IsRemote
-            // Assert.True(ctx.IsRemote);
-
+            Assert.False(ctx.IsRemote);
```
Comment (CodeBlanch) [inline test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs:62]: On the left it says Assert.True but the new version has Assert.False. Should Extract be setting IsRemote = true?

**Comment by eddynaka** [comment]
I was looking at this:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/90c5b3af112bccb04478b50bddfd6973ddcbf633/src/OpenTelemetry.Shims.OpenTracing/SpanContextShim.cs#L44-L48

Can I expose in `SpanContext` the Baggage?

File: test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs (hunk 1 above)
Comment (eddynaka) [reply test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs:62]: That's a good question! would that be in the spec? or is that something from us to decide?

Comment (CodeBlanch) [reply test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs:62]: I think it's safe to just make it true. `IsRemote` means the ActivityContext came from an upstream source. Extract is building the ActivityContext from the upstream source. So an extract context really implies IsRemote=true. At least in my thinking 😄 

IsRemote = false would only be for spans created in-process, using something like ActivitySource.StartActivity.

Actually, the spec does help in this case! From [SpanContext](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#spancontext):
> IsRemote is a boolean which is true if the SpanContext was propagated from a remote parent. When extracting a SpanContext through the Propagators API, its IsRemote flag MUST be set to true, whereas the SpanContext of any child spans MUST have it set to false.

**Comment by CodeBlanch** [comment]
> Can I expose in `SpanContext` the Baggage?

I don't see why not. Maybe it is left out there because we don't actually support the propagation of baggage yet? Defer to @cijothomas on that!

File: test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs (hunk 1 above)
Comment (cijothomas) [reply test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs:62]: If using propagator.Extract - then it should be isRemote=true. The ActivityContext.IsRemote will be true for the extracted context. When an activity is created with this extracted context as its parent, then newly created activities' ActivityContext will NOT have isRemote=true.

Comment (eddynaka) [reply test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs]: @cijothomas / @CodeBlanch , thanks for the comments. Just pushed another change. Let me know if that's right!

**Comment by cijothomas** [comment]
> > Can I expose in `SpanContext` the Baggage?
> 
> I don't see why not. Maybe it is left out there because we don't actually support the propagation of baggage yet? Defer to @cijothomas on that!

Lets address it in separate PR. May be @rajkumar-rangaraj knows something about it. Else we can discuss.

File: test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs [hunk 2]
```diff
@@ -97,9 +94,7 @@ public void TraceContextFormat_IsBlankIfNoHeader()
             var ctx = f.Extract(headers, Getter);
 
             Assert.False(ctx.IsValid());
-
-            // TODO: when ActivityContext supports IsRemote
-            // Assert.True(ctx.IsRemote);
+            Assert.False(ctx.IsRemote);
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs:97]: wondering why are validating this? if context is not valid, (as expected), why validating anything else on it?

Comment (eddynaka) [reply test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs:97]: ok. i can remove those.

Comment (eddynaka) [reply test/OpenTelemetry.Tests/Implementation/Trace/Propagation/TraceContextActivityTest.cs]: just updated.

**Pull request merged**